resources/lib/player_listener.py

```diff
@@ -147,12 +147,12 @@ def onAVStarted(self):  # type: () -> None
 
         self.start_listener()
 
-    def _select_next_checkpoint(self):
+    def _select_next_checkpoint(self, playback_started: bool = False):
         reduce_skips_seconds = (
             addon.get_config(CONF_REDUCE_SKIPS_MS, int) / 1000.0
         )
 
-        current_time = self.getTime()
+        current_time = 0 if playback_started else self.getTime()
```

Comment:
Let's say we have a 3 second segment right at the start and it takes 4 seconds for the sponsorblock api to return the segments. The player has already played past the end of the segment, but because we overwrite the time to 0 here, the addon will actually skip _backwards_ to the end of the segment.

I'm fine with merging this as-is, because this should be a rare occurrence, but I just wanted to leave it as a note.

Reply:
But otherwise we'll see the entire segment, right? We must choose the lesser of two evils :)

Reply:
There is another option though. We can just make sure that we're still _in_ the segment.
Right now the function looks for the next segment ahead of us, but if `playback_started` is true, we just have to look for the segment we're currently in.

Here's a suggestion for how to achieve this:

1. don't do anything with `playback_started` in `_select_next_checkpoint`, instead forward the value to `_is_segment_skippable`.
2. In `_is_segment_skippable`, use the `playback_started` value to accept segments (or in this case _not reject_) we're currently inside of.

The code for this adjusted `_is_segment_skippable` function could look like this:

```python
def _is_segment_skippable(self, seg, current_time, reduce_skips_seconds, *, playback_started: bool = False):
    if seg.end < current_time:
        # this segment is already in the past
        return False
    
    if (not playback_started) and seg.start < current_time:
        # we're already in this segment. If playback just started, we want to skip the segment anyway, but otherwise
        # this means the user manually went back into the segment.
        return False

    chained_end = self.__get_segment_end_handle_overlap(seg)
    min_skip_position = current_time + reduce_skips_seconds 
    if chained_end < min_skip_position:
        logger.debug("skipping segment %s because there is not enough margin for 'Reduce all skips by this much' setting (%g)", seg, reduce_skips_seconds)
        return False

    if chained_end - seg.start < reduce_skips_seconds:
        logger.debug("skipping segment %s because it is shorter than 'Reduce all skips by this much' setting (%g)", seg, reduce_skips_seconds)
        return False

    return True
```

Reply:
edited